docs-app/public/docs/3-components/accordion.md

```diff
@@ -0,0 +1,291 @@
+      }
+    }
+
+    @keyframes slide-up {
+      from {
+        height: 40px;
+      }
+
+      to {
+        height: 0;
+      }
+    }
+  </style>
+</template>
+```
+</details>
+
+## Features
+
+* Full keyboard navigation
+* Can be controlled or uncontrolled
+
+## Installation
+
+```bash
+pnpm add ember-primitives
+```
+
+## Anatomy
+
+```js
+import { Accordion } from 'ember-primitives';
+```
+
+or for non tree-shaking environments:
+```js
+import { Accordion } from 'ember-primitives/components/accordion';
+```
+
+```gjs
+import { Accordion } from 'ember-primitives';
+
+<template>
+  <Accordion as |A|>
+    <A.Item as |I|>
+      <I.Header as |H|>
+        <H.Trigger>Trigger</H.Trigger>
+      </I.Header>
+      <I.Content>Content</I.Content>
+  </Accordion>
+</template>
+```
+
+## API Reference
+
+<details open>
+<summary><h3>Accordion</h3></summary>
+
+```gjs live no-shadow
+import { ComponentSignature } from 'docs-app/docs-support';
+
+<template>
+  <ComponentSignature @module="index" @name="Accordion" />
+</template>
+```
+
+### State Attributes
+| key | description |
+| :---: | :----------- |
+| `data-disabled` | Indicates whether the accordion is disabled. |
+</details>
+
+<details>
+<summary><h3>AccordionItem</h3></summary>
+
+```gjs live no-shadow
+import { ComponentSignature } from 'docs-app/docs-support';
+
+<template>
+  <ComponentSignature @module="index" @name="AccordionItem" />
+</template>
+```
+
+### State Attributes
+| key | description |
+| :---: | :----------- |
+| `data-state` | "open" or "closed", depending on whether the accordion item is expanded or collapsed. |
+| `data-disabled` | Indicates whether the accordion item is disabled. |
+</details>
+
+<details>
+<summary><h3>AccordionHeader</h3></summary>
+
+```gjs live no-shadow
+import { ComponentSignature } from 'docs-app/docs-support';
+
+<template>
+  <ComponentSignature @module="index" @name="AccordionHeader" />
+</template>
+```
+
+### State Attributes
+| key | description |
+| :---: | :----------- |
+| `data-state` | "open" or "closed", depending on whether the accordion item is expanded or collapsed. |
+| `data-disabled` | Indicates whether the accordion item is disabled. |
+</details>
+
+<details>
+<summary><h3>AccordionTrigger</h3></summary>
+
+```gjs live no-shadow
+import { ComponentSignature } from 'docs-app/docs-support';
+
+<template>
+  <ComponentSignature @module="index" @name="AccordionTrigger" />
+</template>
+```
+
+### State Attributes
+| key | description |
+| :---: | :----------- |
+| `data-state` | "open" or "closed", depending on whether the accordion item is expanded or collapsed. |
+| `data-disabled` | Indicates whether the accordion item is disabled. |
+</details>
+
+<details>
+<summary><h3>AccordionContent</h3></summary>
+
+```gjs live no-shadow
+import { ComponentSignature } from 'docs-app/docs-support';
+
+<template>
+  <ComponentSignature @module="index" @name="AccordionContent" />
+</template>
+```
+</details>
+
+## Accessibility
```

Comment:
:raised_hands:

Reply:
however, additional question -- how do we communicate that buttons / accordions are disabled?

I see: https://github.com/universal-ember/ember-primitives/pull/132/files#diff-be8ddb6d68189ac36a0b0aac331f7dbc96013a8310b5ba5036b832540cd25071R37

but, the `disabled` attribute may not be what we want :sweat_smile: 
idk if we can make that call here tho -- it is a little bit of a gray area.

mainly, if a focusable element becomes disabled, you can no longer focus it, and then once you can no longer focus it, you can no longer find out _why_ it's disabled. It's a pattern that the original implementation of `disabled` forgot about, but it's why patterns like this have cropped up in popularity:  https://css-tricks.com/making-disabled-buttons-more-inclusive/


also cc @ynotdraw as we've callab'd on this before as well!
(Do you have advice for how to disable an accordion clickable? -- maybe aria-disabled works for accordions as well! (which would give consumers of this component a chance to explain why the accordion is disabled))

Reply:
Yeah, what I've done/seen is exactly as @NullVoxPopuli describes:

1. Don't use the `disabled` attribute on the trigger
2. Use the [`aria-disabled` attribute](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-disabled) instead
3. Since you'll need to know if the attribute is set in JS/TS, a lot of times it's easier to control this via a component argument like `@isDisabled`.  Then in your click handler, you can "eject" from opening/closing the accordion in the click handler if `@isDisabled` is set to `true`.
4. Provide a mechanism to give feedback as to why the control is disabled - sometimes this is done by only `yield`ing back content and allowing the consumer to pass in whatever they want (ex: `<span class="sr-only">Disabled because ____</span>`, or attaching a tooltip to the element).

There is a 'downside' to using `@isDisabled` over simply allowing the `disabled` attribute - I think it adds a (small) bit of overhead when using the components as people may try to just use the attribute.  But to @NullVoxPopuli 's point, I think trying to push folks to be more accessible-friendly is a good thing.

Reply:
Wow, this is very interesting, thanks for sharing this!

I replaced `disabled` with `aria-disabled="true"/"false"` on the trigger. Seems like I was already checking for `@disabled` in my click handler so I didn't need to make any changes there.

As for users of the component being able to provide feedback to their users on why the item is disabled; I believe they could already do so using the `yield` in either the `Item` component, the `Header` component, or the `Trigger` component.

Let me know if you think this is sufficient

Reply:
Yup, makes sense to me. As long as a yield is somewhere it should be doable 🎉